mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java

```diff
@@ -56,6 +59,30 @@
 public interface KeyContract extends EventBusContract {
 
     interface SingleEventBusKeyContract extends EventBusContract {
+        @Test
+        default void notificationShouldNotExceedRate() {
+            int eventCount = 50;
+            AtomicInteger nbCalls = new AtomicInteger(0);
+            AtomicInteger finishedExecutions = new AtomicInteger(0);
+            AtomicBoolean rateExceeded = new AtomicBoolean(false);
+
+            eventBus().register(event -> {
+                if (nbCalls.get() - finishedExecutions.get() > EventBus.EXECUTION_RATE) {
+                    rateExceeded.set(true);
+                }
+                nbCalls.incrementAndGet();
+                Thread.sleep(Duration.ofMillis(200).toMillis());
+                finishedExecutions.incrementAndGet();
+
+            }, KEY_1);
+
+            IntStream.range(0, eventCount)
+                .forEach(i -> eventBus().dispatch(EVENT, KEY_1).block());
```

Comment:
Wouldn't it be faster/more interesting to use a `Flux` (and do it in parallel)?

Reply:
maybe but the simpler the better, right ? It's just a test

Reply:
ok